pandas/tests/test_nanops.py

```diff
@@ -1002,6 +1003,19 @@ def prng(self):
         return np.random.RandomState(1234)
 
 
+def test_add_overflow():
+    # see gh-14068
+    msg = "too (big|large) to convert"
+    with tm.assertRaisesRegexp(OverflowError, msg):
```

Comment:
so we need to check for a TDI + TD (and DTI + TD), IOW an array + scalar
and TDI + TDI and DTI + DTI (array + array)

also test with NaT's included as some elements of the array (and some overflowing elements)